pycbc/inference/sampler/dynesty.py

```diff
@@ -158,10 +200,24 @@ def from_config(cls, cp, model, output_file=None, nprocesses=1,
         return obj
 
     def checkpoint(self):
-        pass
+        """Checkpoint function for dynesty sampler
+        """
+        self.__getstate__ = self.getstate()
+        pickle_obj = self._sampler
+        for fn in [self.checkpoint_file]:
+            with loadfile(fn, 'a') as fp:
+                fp.write_pickled_data_into_checkpoint_file(pickle_obj)
 
     def resume_from_checkpoint(self):
-        pass
+        for fn in [self.checkpoint_file]:
+            with loadfile(fn, 'r') as fp:
+                self._sampler = fp.read_pickled_data_from_checkpoint_file()
+                self._sampler.rstate = numpy.random
```

Comment:
See paired comment, this won't restore the random state to how it left off when checkpointed.

Reply:
@ahnitz Now I have changed it and using functions written by @cdcapano: `write_random_state` and `read_random_state` to keep track of random state and restore it from previous checkpoint.